cuegui/cuegui/MenuActions.py

```diff
@@ -1565,10 +1565,26 @@ def changeAllocation(self, rpcObjects=None):
                 self._caller, title, body, sorted(allocations.keys()), 0, False)
             if choice:
                 allocation = allocations[str(allocationName)]
+                error_hosts = []
                 for host in hosts:
-                    self.cuebotCall(host.setAllocation,
-                                    "Set Allocation on %s Failed" % host.data.name,
-                                    allocation)
+                    try:
+                        self.cuebotCall(host.setAllocation,
+                                        "Set Allocation on %s Failed" % host.data.name,
+                                        allocation)
+                    except Exception as e:
```

Comment:
Is there a more specific exception type you can catch here?

Reply:
Unfortunately not. The gRPC call returns a Rendevous exception, hence I parse the exception details to check if it's an allocation error

Reply:
I recently stumbled upon a similar grpc exception handling issue and this is a wide spread issue, most locations on the code are catching a general exception to handle grpc exceptions.

A better option is to catch `grpc.RpcError` and check the `.details()` to check the msg.

```python
try:
  something_that_triggers_rendezvous()
except grpc.RpcError as rpc_error_call:  
  code = rpc_error_call.code() # (grpc.StatusCode.UNAVAILABLE, grpc.StatusCode.CANCELLED, ...)
  details = rpc_error_call.details()
```

See for exemple `connectGrpcWithRetries` on rqnetwork.py.

IMO this can be handled in another PR as it's a widespread implementation error.

Reply:
@bcipriano waiting for your confirmation to merge this